src/components/functional/Chart.jsx

```diff
@@ -0,0 +1,149 @@
+"use strict";
+
+import React from 'react';
+
+import TitleWithButton from '../pure/TitleWithButton';
+import SelectableLabelList from '../pure/SelectableLabelList';
+import LoadingIf from '../hoc/LoadingIf'
+import Hideable from '../hoc/Hideable'
+import { AreaChart } from 'react-d3-components';
+import ScatterChart from '../pure/ScatterChartRD3';
+
+const MARGINS = {top: 10, bottom: 50, left: 50, right: 20};
+const CHART_INTERPOLATION = 'step-after';
+const OUTER_STYLE = {display: "flex", marginBottom: 10};
+const SIDE_STYLE = {display: "flex", flexDirection: "column", width: 150};
+const MIDDLE_STYLE = {display: "flex", flexDirection: "column", flex: "auto"};
+
+class Chart extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {selectedElements: new Set()};
+
+        this.WrappedAreaChart = Hideable(AreaChart);
```

Comment:
Maybe `HideableAreaChart` is a better name. `Wrapped` doesn't tell me anything :)